App/AuthProvider.js

```diff
@@ -3,223 +3,126 @@
+const open = require('open');
 
 class AuthProvider {
-  clientApplication;
-  cryptoProvider;
-  authCodeUrlParams;
-  authCodeRequest;
-  pkceCodes;
-  account;
-  customFileProtocolName;
-
-  constructor() {
-    /**
-     * Initialize a public client application. For more information, visit:
-     * https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-node/docs/initialize-public-client-application.md
-     */
-    this.clientApplication = new PublicClientApplication(msalConfig);
-    this.account = null;
-
-    // Initialize CryptoProvider instance
-    this.cryptoProvider = new CryptoProvider();
+    clientApplication;
+    loginRequest;
+    account;
+
+    constructor() {
+        /**
+         * Initialize a public client application. For more information, visit:
+         * https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-node/docs/initialize-public-client-application.md
+         */
+        this.clientApplication = new PublicClientApplication(msalConfig);
+        this.account = null;
+
+        this.setRequestObjects();
+    }
 
     /**
-     * To demonstrate best security practices, this Electron sample application makes use of
-     * a custom file protocol instead of a regular web (https://) redirect URI in order to
-     * handle the redirection step of the authorization flow, as suggested in the OAuth2.0 specification for Native Apps.
+     * Initialize request objects used by this AuthModule.
      */
-    this.customFileProtocolName = REDIRECT_URI.split(":")[0];
-
-    this.setRequestObjects();
-  }
-
-  // Creates a "popup" window for interactive authentication
-  static createAuthWindow() {
-    return new BrowserWindow({
-      width: 400,
-      height: 600,
-    });
-  }
-
-  /**
-   * Initialize request objects used by this AuthModule.
-   */
-  setRequestObjects() {
-    const requestScopes = ["User.Read"];
-
-    this.authCodeUrlParams = {
-      scopes: requestScopes,
-      redirectUri: REDIRECT_URI,
-    };
-
-    this.authCodeRequest = {
-      scopes: requestScopes,
-      redirectUri: REDIRECT_URI,
-      code: null,
-    };
-
-    this.pkceCodes = {
-      challengeMethod: "S256", // Use SHA256 Algorithm
-      verifier: "", // Generate a code verifier for the Auth Code Request first
-      challenge: "", // Generate a code challenge from the previously generated code verifier
-    };
-  }
-
-  async login() {
-    const authResult = await this.getTokenInteractive(this.authCodeUrlParams);
-    return this.handleResponse(authResult);
-  }
-
-  async logout() {
-    if (this.account) {
-      await this.clientApplication.getTokenCache().removeAccount(this.account);
-      this.account = null;
+    setRequestObjects() {
+        const requestScopes = ['User.Read', 'Mail.Read'];
+
+        this.loginRequest = {
+            scopes: requestScopes,
+            redirectUri: REDIRECT_URI,
+            openBrowser: this.openBrowser,
+        };
```

Comment:
```suggestion
        const requestScopes = ['User.Read', 'Mail.Read'];

        this.loginRequest = {
            scopes: requestScopes,
            redirectUri: REDIRECT_URI,
            openBrowser: this.openBrowser,
        };
```

You don't really need to keep track of loginRequest as a class member. It's probably simpler to prepare request object in scope before using acquireTokenInteractive